controller/src/job/mod.rs

```diff
@@ -122,3 +127,116 @@ pub(crate) async fn delete_job(k8s_client: kube::Client, name: &str) -> JobResul
     let _ = result?;
     Ok(())
 }
+
+async fn get_pod(k8s_client: kube::Client, job_name: &str) -> JobResult<String> {
+    let pod_api: Api<Pod> = Api::namespaced(k8s_client, NAMESPACE);
+    let name = pod_api
+        .list(&ListParams {
+            label_selector: Some(format!("job-name={}", job_name)),
+            ..Default::default()
+        })
+        .await
+        .context(error::NotFoundSnafu {})?
+        .items
+        .first()
+        .context(error::NoPodsSnafu {
+            job: job_name.to_string(),
+        })?
+        .name_any();
+
+    Ok(name)
+}
+
+async fn pod_logs(k8s_client: kube::Client, pod_name: &str) -> JobResult<String> {
+    let log_params = LogParams {
+        follow: false,
+        pretty: true,
+        ..Default::default()
+    };
+    let pod_api: Api<Pod> = Api::namespaced(k8s_client, NAMESPACE);
+
+    pod_api
+        .logs(pod_name, &log_params)
+        .await
+        .context(error::NoLogsSnafu { pod: pod_name })
+}
+
+pub(crate) async fn archive_logs(k8s_client: kube::Client, job_name: &str) -> JobResult<()> {
+    let archive_logs = match env::var(TESTSYS_CONTROLLER_ARCHIVE_LOGS) {
+        Ok(s) => s == true.to_string(),
+        Err(e) => {
+            warn!(
+                "Unable to read environment variable '{}': {}",
+                TESTSYS_CONTROLLER_ARCHIVE_LOGS, e
+            );
+            false
+        }
+    };
+
+    if !archive_logs {
+        return Ok(());
+    }
+    let config = aws_config::from_env().load().await;
+    let client = aws_sdk_cloudwatchlogs::Client::new(&config);
+
+    match client
+        .create_log_group()
+        .log_group_name("testsys")
+        .send()
+        .await
+    {
+        Ok(_) => info!("Creating log group"),
+        Err(e) => {
+            let service_error = e.into_service_error();
+            if service_error.is_resource_already_exists_exception() {
+                info!("Log group already exists.")
```

Comment:
Maybe a little noisy?

```suggestion
                debug!("testsys Cloudwatch Log group already exists.")
```